src/components/PostDetails.tsx

```diff
@@ -1,117 +1,133 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
-import { NewCommentForm } from './NewCommentForm';
 
-export const PostDetails: React.FC = () => {
-  return (
-    <div className="content" data-cy="PostDetails">
-      <div className="content" data-cy="PostDetails">
-        <div className="block">
-          <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
-          </h2>
+import { Comment } from '../types/Comment';
+import { Post } from '../types/Post';
+import { getComment, deleteComment } from '../api/ApiMethods';
+import { NewCommentForm } from './NewCommentForm';
 
-          <p data-cy="PostBody">
-            eveniet quo quis
-            laborum totam consequatur non dolor
-            ut et est repudiandae
-            est voluptatem vel debitis et magnam
-          </p>
-        </div>
+type Props = {
+  openPost: Post | null,
+  setOpenForm: (v: boolean) => void,
+  openForm: boolean,
+};
 
-        <div className="block">
-          <Loader />
+export const PostDetails: React.FC<Props> = ({
+  openPost,
+  setOpenForm,
+  openForm,
+}) => {
+  const [postComments, setPostComments] = useState<Comment[]>([]);
+  const [hasError, setHasError] = useState(false);
+  const [loading, setLoading] = useState(false);
 
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
+  const handleDeleteComments = (comId: number) => {
+    deleteComment(comId)
+      .then(() => {
+        const updatePostComments = [...postComments];
+        const index = updatePostComments.findIndex(com => com.id === comId);
 
-          <p className="title is-4" data-cy="NoCommentsMessage">
-            No comments yet
-          </p>
+        updatePostComments.splice(index, 1);
+        setPostComments(updatePostComments);
+      })
+      .catch(() => setHasError(true));
+  };
 
-          <p className="title is-4">Comments:</p>
+  useEffect(() => {
+    if (openPost) {
+      setLoading(true);
+      setHasError(false);
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
+      getComment(openPost.id)
+        .then(PostComments => setPostComments(PostComments))
```

Comment:
```suggestion
        .then(costComments => setPostComments(costComments))
```
camelCase for variable names (that's not a class, React component or type, etc)